dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java

```diff
@@ -111,12 +113,19 @@ public CWLClient(AbstractEntryClient abstractEntryClient) {
 
         fileProvisioning = new FileProvisioning(abstractEntryClient.getConfigFile());
 
-        // Set the launcher
-        INIConfiguration config = Utilities.parseConfig(abstractEntryClient.getConfigFile());
-        cwlLauncherType = config.getString(CWL_RUNNER, DEFAULT_LAUNCHER);
+        if (!abstractEntryClient.isWesCommand()) {
+            // Set the launcher
+            INIConfiguration config = Utilities.parseConfig(abstractEntryClient.getConfigFile());
+            cwlLauncherType = config.getString(CWL_RUNNER, DEFAULT_LAUNCHER);
+        } else {
+            cwlLauncherType = WES;
+        }
 
         BaseLauncher launcher;
         switch (cwlLauncherType) {
+        case WES:
```

Comment:
I still maintain as I have commented previously, and still without having looked at the code in depth, that you should be able to have the WES launcher code not require changes to WDLClient or CWLClient (or whatever language that we add support for that WES supports).

On the other hand, you're not changing very much, so maybe it's not worth it. Maybe we can take a quick look at tomorrow. 

I could also be totally off-base. :)

Reply:
Not sure how this could be done.

Reply:
Took a quick look, and I think you could have `AbstractEntryClient.launch()` call a new method `AbstractEntryClient.launchWes()`, where `launchWes()` basically has this code. Maybe I'm missing something, but it seems to me there is no WES code in here that needs to know anything about CWLClient, nor does any of the WES code need to know anything about WDLClient.

That way, when/if WES supports new languages, e.g., NextFlow, Galaxy, etc., we would only have to make changes in one place (or maybe not even -- it may just work), rather than in each language implementation.

That said, this code works, so that can be done as a later improvement.

Reply:
https://ucsc-cgl.atlassian.net/browse/DOCK-572